test/e2e/network/loadbalancer.go

```diff
@@ -1236,6 +1244,134 @@ var _ = common.SIGDescribe("LoadBalancers", func() {
+
+		nodeNames := e2edaemonset.SchedulableNodes(cs, ds)
+		e2eskipper.SkipUnlessAtLeast(len(nodeNames), 2, "load-balancer rolling update test requires at least 2 schedulable nodes for the DaemonSet")
+
+		ginkgo.By(fmt.Sprintf("Creating DaemonSet %q", name))
+		ds, err = cs.AppsV1().DaemonSets(ns).Create(context.TODO(), ds, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+
+		ginkgo.By("Checking that daemon pods launch on every schedulable node of the cluster")
+		err = wait.PollImmediate(dsRetryPeriod, dsRetryTimeout, e2edaemonset.CheckDaemonPodOnNodes(f, ds, nodeNames))
+		framework.ExpectNoError(err, "error waiting for daemon pods to start")
+		err = e2edaemonset.CheckDaemonStatus(f, name)
+		framework.ExpectNoError(err)
+
+		ginkgo.By(fmt.Sprintf("Creating a service %s with type=LoadBalancer in namespace %s", name, ns))
+		jig := e2eservice.NewTestJig(cs, ns, name)
+		jig.Labels = labels
+		service, err := jig.CreateLoadBalancerService(e2eservice.GetServiceLoadBalancerCreationTimeout(cs), func(svc *v1.Service) {})
+		framework.ExpectNoError(err)
+
+		lbNameOrAddress := e2eservice.GetIngressPoint(&service.Status.LoadBalancer.Ingress[0])
+		svcPort := int(service.Spec.Ports[0].Port)
+
+		ginkgo.By("Hitting the DaemonSet's pods through the service's load balancer")
+		timeout := e2eservice.LoadBalancerLagTimeoutDefault
+		if framework.ProviderIs("aws") {
+			timeout = e2eservice.LoadBalancerLagTimeoutAWS
+		}
+		e2eservice.TestReachableHTTP(lbNameOrAddress, svcPort, timeout)
+
+		ginkgo.By("Starting a goroutine to continuously hit the DaemonSet's pods through the service's load balancer")
+		var totalRequests uint64 = 0
+		var failedRequests uint64 = 0
+		done := make(chan struct{})
+		defer close(done)
+		go func() {
+			defer ginkgo.GinkgoRecover()
+
+			wait.Until(func() {
+				atomic.AddUint64(&totalRequests, 1)
+				client := &http.Client{
+					Transport: utilnet.SetTransportDefaults(&http.Transport{
+						DisableKeepAlives: true,
+					}),
+					Timeout: 5 * time.Second,
+				}
+				ipPort := net.JoinHostPort(lbNameOrAddress, strconv.Itoa(svcPort))
+				msg := "hello"
+				url := fmt.Sprintf("http://%s/echo?msg=%s", ipPort, msg)
+				resp, err := client.Get(url)
+				if err != nil {
+					framework.Logf("Got error testing for reachability of %s: %v", url, err)
+					atomic.AddUint64(&failedRequests, 1)
+					return
+				}
+				defer resp.Body.Close()
+				if resp.StatusCode != http.StatusOK {
+					framework.Logf("Got bad status code: %d", resp.StatusCode)
+					atomic.AddUint64(&failedRequests, 1)
+					return
+				}
+				body, err := io.ReadAll(resp.Body)
+				if err != nil {
+					framework.Logf("Got error reading HTTP body: %v", err)
+					atomic.AddUint64(&failedRequests, 1)
+					return
+				}
+				if string(body) != msg {
+					framework.Logf("The response body does not contain expected string %s", string(body))
+					atomic.AddUint64(&failedRequests, 1)
+					return
+				}
+			}, time.Duration(0), done)
```

Comment:
is not this very strict? 

interesting anecdote, is that with a similar test a team exhausted all the cloud gw ephemeral ports (you are disabling keepalives) and it took a week to figure out why the test failed 

can we use 1*Second?

Reply:
The idea would be to have the go routine continuously do HTTP requests to the load balancer to see how much traffic is dropped during rolling updates.

It simulates a production Kubernetes workload with a lot of unique clients to it (that's why disabling keepalives makes sense). And a rolling update needs to be done. How much traffic is lost during the operation ? Ideally, we should have zero downtime (which is what `ProxyTerminatingEndpoints` feature gate is supposed to do, I believe). 

I can update the polling interval to `1 * Second` if you want, but I believe that `time.Duration(0)` works for this use case.

Reply:
0 fair enough, thanks for the detailed explanation